rkt/api_service_test.go

```diff
@@ -0,0 +1,466 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema"
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/appc/spec/schema/types"
+	"github.com/coreos/rkt/api/v1alpha"
+)
+
+func TestFilterPod(t *testing.T) {
+	tests := []struct {
+		pod      *v1alpha.Pod
+		manifest *schema.PodManifest
+		filter   *v1alpha.PodFilter
+		filtered bool
+	}{
+		// Has the status.
+		{
+			&v1alpha.Pod{
+				State: v1alpha.PodState_POD_STATE_RUNNING,
+			},
+			&schema.PodManifest{},
+			&v1alpha.PodFilter{
+				States: []v1alpha.PodState{v1alpha.PodState_POD_STATE_RUNNING},
+			},
+			false,
+		},
+		// Doesn't have the status.
+		{
+			&v1alpha.Pod{
+				State: v1alpha.PodState_POD_STATE_EXITED,
+			},
+			&schema.PodManifest{},
+			&v1alpha.PodFilter{
+				States: []v1alpha.PodState{v1alpha.PodState_POD_STATE_RUNNING},
+			},
+			true,
+		},
+		// Has the app name.
+		{
+			&v1alpha.Pod{
+				Apps: []*v1alpha.App{{Name: "app-foo"}},
+			},
+			&schema.PodManifest{},
+			&v1alpha.PodFilter{
+				AppNames: []string{"app-foo"},
```

Comment:
Can you try:
```
AppNames: []string{"app-first", "app-foo", "app-last"},
```
To check it is not filtered when it matches only one of the 3 given app names?

Ditto for network and annotation.